#! /bin/sh /usr/share/dpatch/dpatch-run
## 04_lzma_fix.patch by <gebi@grml.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: support for LZMA compression

@DPATCH@

diff --git a/squashfs-tools/mksquashfs.c b/squashfs-tools/mksquashfs.c
index 10f14eb..276f680 100644
--- a/squashfs-tools/mksquashfs.c
+++ b/squashfs-tools/mksquashfs.c
@@ -898,6 +898,13 @@ unsigned int mangle2(z_stream **strm, char *d, char *s, int size,
 		case SZ_ERROR_PARAM:
 			BAD_ERROR("lzma::compress failed, invalid parameters\n");
 			break;
+        case SZ_ERROR_OUTPUT_EOF:
+            memcpy(d, s, size);
+            return size | (data_block ? SQUASHFS_COMPRESSED_BIT_BLOCK :
+                    SQUASHFS_COMPRESSED_BIT);
+        case SZ_ERROR_THREAD:
+			BAD_ERROR("lzma::compress failed, errors in multithreading functions\n");
+			break;
 		/* should not happen */
 		default:
 			BAD_ERROR("lzma::compress failed, unknown error\n");
